commands/command_migrate_info.go

```diff
@@ -0,0 +1,161 @@
+	}
+
+	migrateInfoAbove = above
+
+	migrate(cmd, args, func(path string, b *odb.Blob) (*odb.Blob, error) {
+		ext := fmt.Sprintf("*%s", filepath.Ext(path))
+
+		if len(ext) > 1 {
+			entry := exts[ext]
+			if entry == nil {
+				entry = &MigrateInfoEntry{Qualifier: ext}
+			}
+
+			entry.Total++
+			entry.BytesTotal += b.Size
+
+			if b.Size > int64(migrateInfoAbove) {
+				entry.TotalAbove++
+				entry.BytesAbove += b.Size
+			}
+
+			// TODO(@ttaylorr): needed?
+			exts[ext] = entry
+		}
+
+		return b, nil
+	})
+
+	entries := EntriesBySize(MapToEntries(exts))
+	sort.Sort(sort.Reverse(entries))
+
+	migrateInfoTopN = tools.ClampInt(migrateInfoTopN, len(entries), 0)
+
+	entries = entries[:tools.MaxInt(0, migrateInfoTopN)]
+
+	entries.Print(os.Stderr)
+}
+
+// MigrateInfoEntry represents a tuple of filetype to total size taken by that
+// file type.
+type MigrateInfoEntry struct {
+	// Qualifier is the filepath's extension.
+	Qualifier string
+
+	BytesAbove int64
+	TotalAbove int64
+	BytesTotal int64
+	Total      int64
+}
+
+// MapToEntries creates a set of `*MigrateInfoEntry`'s for a given map of
+// filepath extensions to file size in bytes.
+func MapToEntries(exts map[string]*MigrateInfoEntry) []*MigrateInfoEntry {
+	entries := make([]*MigrateInfoEntry, 0, len(exts))
+	for _, entry := range exts {
+		entries = append(entries, entry)
+	}
+
+	return entries
+}
+
+// EntriesBySize is an implementation of sort.Interface that sorts a set of
+// `*MigrateInfoEntry`'s
+type EntriesBySize []*MigrateInfoEntry
+
+// Len returns the total length of the set of `*MigrateInfoEntry`'s.
+func (e EntriesBySize) Len() int { return len(e) }
+
+// Less returns the whether or not the MigrateInfoEntry given at `i` takes up
+// less total size than the MigrateInfoEntry given at `j`.
+func (e EntriesBySize) Less(i, j int) bool { return e[i].BytesAbove < e[j].BytesAbove }
+
+// Swap swaps the entries given at i, j.
+func (e EntriesBySize) Swap(i, j int) { e[i], e[j] = e[j], e[i] }
+
+// Print formats the `*MigrateInfoEntry`'s in the set and prints them to the
+// given io.Writer, "to", returning "n" the number of bytes written, and any
+// error, if one occurred.
+func (e EntriesBySize) Print(to io.Writer) (int, error) {
+	extensions := make([]string, 0, len(e))
+	for _, entry := range e {
+		extensions = append(extensions, entry.Qualifier)
+	}
+	extensions = tools.Ljust(extensions)
+
+	files := make([]string, 0, len(e))
+	for _, entry := range e {
+		bytes := humanize.FormatBytes(uint64(entry.BytesAbove))
+		above := entry.TotalAbove
+		total := entry.Total
+
+		file := fmt.Sprintf("%s, %d/%d files(s)",
+			bytes, above, total)
+
+		files = append(files, file)
+	}
+	files = tools.Rjust(files)
+
+	percentages := make([]string, 0, len(e))
+	for _, entry := range e {
+		percentAbove := 100 * (float64(entry.TotalAbove) / float64(entry.Total))
+
+		percentage := fmt.Sprintf("%.0f%%", percentAbove)
+
+		percentages = append(percentages, percentage)
+	}
```

Comment:
Two minor things about this: 

First, it's given an `io.Writer`, but the output is buffered. It's not exactly a lot of data, but I'm just used to writing it out with `fmt.Fprintf` if the output doesn't need post-processing. However, having to deal with just 1 `err` from `fmt.Fprintln` is kinda nice.

Second: You could reduce how often `e` is looped by building `extensions`, `files`, and `percentages` at the same time. At a high level, the structure of this func would be something like:

1. Init `extensions`, `files`, `percentages`
2. Loop through `e`, appending to those slices
3. `Rjust` or `Ljust` them
4. Write header
5. Write each line with the justified `extensions`, `files`, `percentages`

I think structurally it makes more sense, but 36179dee6c1cc9eb2d1206d54360d39f306ba3e4 saves a whopping 3 LOC (7 if I didn't add comments).